### PR TITLE
Refactoring for cross-platform xdp placeholder

### PR DESCRIPTION
**Pull request by ami-GS**
## Description
refactoring datapath_raw_* to make Linux XDP placeholder

## Testing
Let's see pipeline.....

## Documentation
N/A

### CODE REVIEW

File: src/platform/datapath_raw_socket.c [hunk 1]
```diff
@@ -1433,7 +769,6 @@ CxPlatFramingWriteHeaders(
                         sizeof(Route->LocalAddress.Ipv6.sin6_addr),
                         IPPROTO_UDP,
                         (uint8_t*)UDP, sizeof(UDP_HEADER) + Buffer->Length);
-                UDP->Checksum = UDP->Checksum != 0 ? UDP->Checksum : ~0;
```
Comment (ami-GS) [inline src/platform/datapath_raw_socket.c:1436]: back

File: src/platform/datapath_raw.h [hunk 2]
```diff
@@ -394,3 +382,100 @@ CxPlatFramingWriteHeaders(
     _In_ uint32_t TcpAckNum,
     _In_ uint8_t TcpFlags
     );
+
+
+//
+// Ethernet / IP Framing Logic
+//
+
+// #pragma pack(push)
```
Comment (ami-GS) [inline src/platform/datapath_raw.h:391]: back

File: src/platform/datapath_raw.h [hunk 3]
```diff
@@ -394,3 +382,100 @@ CxPlatFramingWriteHeaders(
...
+
+#define IPV4_VERSION 4
+#define IPV6_VERSION 6
+#define IPV4_VERSION_BYTE (IPV4_VERSION << 4)
+#define IPV4_DEFAULT_VERHLEN ((IPV4_VERSION_BYTE) | (sizeof(IPV4_HEADER) / sizeof(uint32_t)))
+
+#define IP_DEFAULT_HOP_LIMIT 128
+
+#define ETHERNET_TYPE_IPV4 0x0008
+#define ETHERNET_TYPE_IPV6 0xdd86
```
Comment (ami-GS) [inline src/platform/datapath_raw.h:481]: last line

File: src/platform/datapath_raw_socket_win.c [hunk 4]
```diff
@@ -0,0 +1,1360 @@
...
+    if (Status != ERROR_IO_PENDING && Status != ERROR_SUCCESS) {
+        Callback(Context, NULL, PathId, FALSE);
+    }
+
+    if (Status == ERROR_IO_PENDING) {
+        return QUIC_STATUS_PENDING;
+    } else {
+        return HRESULT_FROM_WIN32(Status);
+    }
+<<<<<<< HEAD
```
Comment (nibanks) [inline src/platform/datapath_raw_socket_win.c:244]: Build error from this

File: src/platform/datapath_raw_socket_win.c [hunk 5]
```diff
@@ -0,0 +1,1360 @@
...
+#else // _WIN32
+    return QUIC_STATUS_NOT_SUPPORTED;
+#endif // _WIN32
+}
+
+//
+// Ethernet / IP Framing Logic
+//
+
+#pragma pack(push)
```
Comment (nibanks) [inline src/platform/datapath_raw_socket_win.c:254]: You already put these in the header file. Remove them from here too.

File: src/platform/datapath_raw_socket_linux.c [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+CxPlatTryAddSocket(
+    _In_ CXPLAT_SOCKET_POOL* Pool,
+    _In_ CXPLAT_SOCKET* Socket
+    )
+{
+    UNREFERENCED_PARAMETER(Pool);
+    UNREFERENCED_PARAMETER(Socket);
+
+    return QUIC_STATUS_NOT_SUPPORTED;
+}
```
Comment (nibanks) [inline src/platform/datapath_raw_socket_linux.c:84]: Let's make sure all files have trailing new lines.

**Review by nibanks — APPROVED** [review]

**Pull request merged**